doc/sphinx/practical-tools/coqide.rst

```diff
@@ -5,130 +5,183 @@
+- Fully check the document (gears icon) - for async mode
+- Previous occurrence (left arrow icon) - find the previous occurrence
+  of the current word (The current word is determined by the cursor position.)
+- Next occurrence (right arrow icon) - find the next occurrence
+  of the current word
+
+.. _asyncmode:
+
+Asynchronous mode
+-----------------
+
+Asynchronous mode uses multiple Coq processes to process elements of the script
+in parallel.  For example, multiple theorems may be processed in parallel.  This is
+described in detail in :ref:`asynchronousandparallelproofprocessing`.
+
+While synchronous mode stops processing at the first error it encounters, in async
+mode, errors only stop processing the proof the error appears in.
+Therefore it can report errors in multiple proofs without manual intervention.
+In addition, async mode lets the user edit failed proofs without invalidating
+successful proofs that appear after it in the script.  The editable part of the proof
+is the part that appears between `Proof.` and `Qed.`.
+
+.. image:: ../_static/async-mode.png
+   :alt: Async mode
+
+In the screenshot, the proof of the failed theorem can be edited (between `Proof.`
+and `Qed.`) without invalidating the theorems that follow it.  The modified
+proof can then be reprocessed using the usual navigation operations.  The light blue
+highlight in the script indicates commands that haven't been processed.
+
+Async mode defers the final type checking step of proofs, leaving the `Qed.` marked in
+a slightly different shade of light blue to indicate this.  To complete the final
+checking, click on the "gears" button on the toolbar ("Fully check the document").
 
 Commands and templates
 ----------------------
 
 The Templates menu allows using shortcuts to insert
-commands. This is a nice way to proceed if you are not sure of the
+commands. This is a nice way to proceed if you're not sure of the
 syntax of the command you want.
 
 Moreover, from this menu you can automatically insert templates of complex
 commands like ``Fixpoint`` that you can conveniently fill afterwards.
 
 Queries
-------------
+-------
 
 .. image:: ../_static/coqide-queries.png
    :alt: CoqIDE queries
 
-We call *query* any command that does not change the current state,
-such as ``Check``, ``Search``, etc. To run such commands interactively, without
-writing them in scripts, CoqIDE offers a *query pane*. The query pane can be
-displayed on demand by using the ``View`` menu, or using the shortcut ``F1``.
-Queries can also be performed by selecting a particular phrase, then choosing an
-item from the ``Queries`` menu. The response then appears in the message window.
-The image above shows the result after selecting of the phrase
-``Nat.mul`` in the script window, and choosing ``Print`` from the ``Queries``
-menu.
+A *query* is any command that does not change the current state, such as
+:cmd:`Check`, :cmd:`Search`, etc.  The *query pane* lets you run such commands
```

Comment:
```suggestion
:cmd:`About`, :cmd:`Check`, `cmd:`Print`, :cmd:`Search`, etc.  The *query pane* lets you run such commands
```

Reply:
Got it.  Do we still need the "etc."?

Reply:
Technically there are a few more so I would keep it.